frontend/src/app/main/ui/shapes/custom_stroke.cljs

```diff
@@ -286,5 +311,56 @@
        child]
 
       :else
-      child)))
-
+      [:*
+       [:g.stroke-shape
+        [:defs
+         [:& stroke-defs {:shape shape :render-id render-id :index index}]]
+        child]])))
+
+(defn build-stroke-props [position shape child value]
+  (let [render-id    (mf/use-ctx muc/render-ctx)
+        url-fill?    (or (some? (:fill-image shape))
+                         (= :image (:type shape))
+                         (> (count (:fills shape)) 1)
+                         (some :fill-color-gradient (:fills shape)))
+        one-fill?    (= (count (:fills shape)) 1)
+        no-fills?    (= (count (:fills shape)) 0)
+        last-stroke? (not= position (- (count (:strokes shape)) 1))
+
+        props        (-> (obj/get child "props")
+                         (obj/clone))
+
+        props (cond
+                (or last-stroke? (and (not url-fill?) no-fills?))
+                (-> props
+                    (obj/without ["fill" "fillOpacity"]))
+
+                url-fill?
+                (obj/set! props "fill" (str "url(#fill-" render-id ")"))
+
+                (and one-fill? (not url-fill?))
+                (obj/merge!
+                 props
+                 (attrs/extract-fill-attrs (get-in shape [:fills 0]) render-id 0)))
+
+        props (-> props
+                  (add-style
+                   (obj/get (attrs/extract-stroke-attrs value position) "style")))]
+    props))
+
+(mf/defc shape-custom-strokes
+  {::mf/wrap-props false}
+  [props]
+  (let [child (obj/get props "children")
+        shape (obj/get props "shape")
+        elem-name (obj/get child "type")]
+    [:*
```

Comment:
This `[:*` is only necessary in the `for` branch. I think it's better to move it together with the nested for.